nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/v2/AbstractAwsProcessor.java

```diff
@@ -0,0 +1,451 @@
+                    .build());
+        }
+
+        return results;
+    }
+
+    protected void configureClientBuilder(final U clientBuilder, final ProcessContext context) {
+        clientBuilder.overrideConfiguration(builder -> builder.putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_PREFIX, DEFAULT_USER_AGENT));
+        clientBuilder.overrideConfiguration(builder -> builder.retryPolicy(RetryPolicy.none()));
+        clientBuilder.httpClient(createSdkHttpClient(context));
+
+        final Region region = getRegion(context);
+        if (region != null) {
+            clientBuilder.region(region);
+        }
+        configureEndpoint(context, clientBuilder);
+
+        final AwsCredentialsProvider credentialsProvider = getCredentialsProvider(context);
+        clientBuilder.credentialsProvider(credentialsProvider);
+    }
+
+    /**
+     * Creates the AWS SDK client.
+     * @param context The process context
+     * @return The created client
+     */
+    @Override
+    public T createClient(final ProcessContext context) {
+        final U clientBuilder = createClientBuilder(context);
+        this.configureClientBuilder(clientBuilder, context);
+        return clientBuilder.build();
+    }
+
+    /**
+     * Creates an AWS service client from the context or returns an existing client from the cache
+     * @param context The process context
+     * @param  awsClientDetails details of the AWS client
+     * @return The created client
+     */
+    protected T getClient(final ProcessContext context, final AwsClientDetails awsClientDetails) {
+        return this.awsClientCache.getOrCreateClient(context, awsClientDetails, this);
+    }
+
+    protected T getClient(final ProcessContext context) {
+        final AwsClientDetails awsClientDetails = new AwsClientDetails(getRegion(context));
+        return getClient(context, awsClientDetails);
+    }
+
+    private SdkHttpClient createSdkHttpClient(final ProcessContext context) {
+        final ApacheHttpClient.Builder builder = ApacheHttpClient.builder();
+
+        final int communicationsTimeout = context.getProperty(TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS).intValue();
+        builder.connectionTimeout(Duration.ofMillis(communicationsTimeout));
+        builder.socketTimeout(Duration.ofMillis(communicationsTimeout));
+        builder.maxConnections(context.getMaxConcurrentTasks());
+
+        if (this.getSupportedPropertyDescriptors().contains(SSL_CONTEXT_SERVICE)) {
+            final SSLContextService sslContextService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+            if (sslContextService != null) {
+                final TrustManager[] trustManagers = new TrustManager[] { sslContextService.createTrustManager() };
+                final KeyManagerFactory keyManagerFactory;
+                try {
+                    keyManagerFactory = KeyStoreUtils.loadKeyManagerFactory(sslContextService.createTlsConfiguration());
```

Comment:
It looks like AWS has a `FileSToreTlsKeyManagersProvider` that could be used. That should avoid the need for KeyStoreUtils and `nifi-security-utils`.